src/redux/middlewares/localStorageUserMiddleware.test.js

```diff
@@ -0,0 +1,80 @@
+import configureStore from "redux-mock-store";
+
+import localStorageUserMiddleware from "./localStorageUserMiddleware";
+
+describe(`localStorageUserMiddleware`, () => {
+  let store;
+  let next = jest.fn();
+  let setUser = jest.fn();
+  let user = { user: "some-user" };
+  const middleware = [localStorageUserMiddleware(setUser)];
+  const mockStore = configureStore({
+    reducer: setUser,
+    middleware,
+  });
+
+  describe("as a cold user", () => {
+    beforeEach(() => {
+      const initialState = {
+        editor: {},
+        auth: {},
+      };
+      store = mockStore(initialState);
+      store.dispatch = jest.fn();
+    });
+
+    describe(`when authKey is set`, () => {
+      beforeEach(() => {
+        localStorage.setItem("authKey", "some-key");
+        localStorage.setItem("some-key", JSON.stringify(user));
+      });
+
+      it(`expects setUser to be called`, () => {
+        localStorageUserMiddleware(setUser)(store)(next)({
+          type: "editor/someAction",
+        });
+        expect(setUser).toBeCalledWith(user);
+      });
+    });
+
+    describe(`when authKey is not set`, () => {
+      beforeEach(() => {
+        localStorage.removeItem("authKey");
+      });
+
+      it(`expects setUser to not be called`, () => {
+        localStorageUserMiddleware(setUser)(store)(next)({
+          type: "editor/someAction",
+        });
+        expect(setUser).not.toBeCalled();
+      });
+    });
+  });
+
+  describe("as a logged in user", () => {
+    const updatedUser = { user: "updated-user" };
+
+    beforeEach(() => {
+      const initialState = {
+        editor: {},
+        auth: { user },
+      };
+      store = mockStore(initialState);
+      store.dispatch = jest.fn();
+    });
+
+    describe(`when authKey has been updated`, () => {
+      beforeEach(() => {
+        localStorage.setItem("authKey", "some-key");
+        localStorage.setItem("some-key", JSON.stringify(updatedUser));
+      });
```

Comment:
Not sure I understand in what circumstances the auth key would ever be updated - I think it's pretty standard if I'm not wrong 🤔

Reply:
yea sorry, badly named. This is testing when the authKey value has changed so in this case "some-key" 😅

Reply:
Ohhhh, okay, so you mean the local storage user has been updated rather than the auth key?

Reply:
yes! That's 100% a better way of putting it 😅 I'll update that!